scipy/stats/_hypotests.py

```diff
@@ -0,0 +1,118 @@
+        The points (t1, ..., tn) where the empirical characterisitc function is
+        to be evaluated. It should a positive distinct numbers. The default
+        value (0.4, 0.8) is proposed in [1]_. Input must not have more than
+        one dimension.
+
+    Returns
+    -------
+    res : tuple
+        The value of the test statistic and the associated p-value.
+
+    See Also
+    --------
+    ks_2samp, anderson_ksamp
+
+    Notes
+    -----
+    Testing whether two samples are generated by the same underlying
+    distribution is a classical question in statistics. A widely used test is
+    the Kolmogorov-Smirnov (KS) test which relies on the empirical
+    distribution function. Epps and Singleton introduce a test based on the
+    empirical characteristic function in [1]_.
+
+    One advantage of the Singleton-Epps test compared to the KS test is that
+    is does not assume a continuous distribution. In [1]_, the authors
+    conclude that the test also has a higher power than the KS test in many
+    examples.
+
+    The p-value is computed from the asymptotic distribution of the test
+    statistic which follows a `chi2` distribution. If the sample size of both
+    `x` and `y` is below 25, the small sample correction proposed in [1]_ is
+    applied to the test statistic.
+
+    The default values of `t` are determined in [1]_ by considering
+    various distributions and finding good values that lead to a high power
+    of the test in general. Table III in [1]_ gives the optimal values for
+    the distributions tested in that study.
+
+    References
+    ----------
+    .. [1] T. W. Epps and K. J. Singleton, "An omnibus test for the two-sample
+       problem using the empirical characteristic function", Journal of
+       Statistical Computation and Simulation 26, p. 177–-203, 1986.
+
+    .. [2] S. J. Goerg and J. Kaiser, "Nonparametric testing of distributions -
+       the Epps–Singleton two-sample test using the empirical characteristic
+       function", The Stata Journal 9(3), p. 454--465, 2009.
+
+    """
+
+    x, y, t = np.asarray(x), np.asarray(y), np.asarray(t)
+    if x.ndim > 1:
+        raise ValueError('x must be 1d, but x.ndim equals {}.'.format(x.ndim))
+    if y.ndim > 1:
+        raise ValueError('y must be 1d, but y.ndim equals {}.'.format(y.ndim))
+    if t.ndim > 1:
+        raise ValueError('t must be 1d, but t.ndim equals {}.'.format(t.ndim))
+
+    t = t[np.nonzero(t)]
+    if len(t) == 0:
+        raise ValueError('t must contain at least one non-zero element.')
+    nx = len(x)
+    if nx < 5:
+        raise ValueError('x should have at least 5 elements, but len(x) '
+                         'equals {}.'.format(nx))
+    ny = len(y)
+    if ny < 5:
+        raise ValueError('y should have at least 5 elements, but len(y) '
+                         'equals {}.'.format(ny))
+    n = nx + ny
+
+    # rescale t with semi-iqr as proposed in [1]
+    sigma = iqr(np.hstack((x, y))) / 2
```

Comment:
if either or `x` or `y` here has a `None` value the iqr() will throw a runtime warning. Does it make sense to check in this function and throw an error here instead?

Reply:
Tests with `nan` and `inf` would be useful to add. I just checked `kstest`, and it happily accepts `inf` while choking on `nan` (not by design I think).

Reply:
do you mean checks in this function (e.g. raising an error if there are non-finite values)? maybe i should add `nan_policy` as well.

Reply:
At the moment, nans and inf values trigger "LinAlgError: SVD did not converge", which is not helpful for the user